pkg/jobs/jobsprofiler/profiler.go

```diff
@@ -57,3 +62,30 @@ func StorePlanDiagram(
 			jobID, err.Error())
 	}
 }
+
+// StorePerNodeProcessorProgressFraction stores the progress fraction for each
+// node and processor executing as part of the job's DistSQL flow.
+func StorePerNodeProcessorProgressFraction(
+	ctx context.Context,
+	db isql.DB,
+	jobID jobspb.JobID,
+	perComponentProgress map[execinfrapb.ComponentID]float32,
+) {
+	if err := db.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
```

Comment:
i think i know what you meant. Doing it this way means we could see the progress regress because the goroutines could be executed out of order. I'll think of how to fix this.

Reply:
Okay, I think this should prevent regressions. There may be an optimization where we don't update the progress if our channel has future values but I think thats more complexity that required at the moment?